flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/PartitionTestUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.io.disk.iomanager.IOManager;
+import org.apache.flink.runtime.taskmanager.TaskActions;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * This class should consolidate all mocking logic for ResultPartitions.
+ * While using Mockito internally (for now), the use of Mockito should not
+ * leak out of this class.
+ */
+public class PartitionTestUtils {
```

Comment:
Could you drop mockito usage from this class? You are not only consolidating the mockito usage, but also later you are introducing new tests that are using this methods.

I think all of those `mock`'ed classes, are quite easy to construct manually (for example `new ResultPartitionManager()`, `IOManager` or `new NoOpTaskActions()`).

Reply:
Can add a later cleanup commit here. Did not want to go too deep down the mockito hole when rearranging this.

Reply:
Using a proper IOManager is tricky, because it needs I/O Worker Thread shutdown. We could use a test mock here that does not start threads.

Reply:
I would still prefer not to expand the mockito usage. Adding a trivial no-op implementation of `IOManager` is easy and we already have one that throws exceptions (`TestIOManager`).  Rest of the fields are already taken care of:

```
		return new ResultPartition(
			"TestTask",
			new NoOpTaskActions(),
			new JobID(),
			new ResultPartitionID(),
			PIPELINED,
			1,
			1,
			new ResultPartitionManager(),
			new NoOpResultPartitionConsumableNotifier(),
			new NoOpIOManager(),
			false);
```